### PR TITLE
Gherkin updation for topology and pipeline

### PR DESCRIPTION
**Pull request by sanketpathak**
Story: https://issues.redhat.com/browse/ODC-5905, https://issues.redhat.com/browse/ODC-5951

### CODE REVIEW

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature [hunk 1]
```diff
@@ -64,6 +64,8 @@ Feature: Create the pipeline from builder page
                   | p-three-1     | kn        |
 
 
+        @un-verified
```
Comment (gajanan-more) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature:67]: What is this tag for?

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature [hunk 2]
```diff
@@ -73,11 +75,12 @@ Feature: Create the pipeline from builder page
...
                   | pipeline_name | task_name        | resource_type | resource_name |
                   | p-git         | openshift-client | Git           | git repo      |
                   | p-img         | buildah          | Image         | image repo    |
-                  | p-storage     | task-storage     | Storage       | storage repo  |
-                  | p-cluster     | task-cluster     | Cluster       | cluster repo  |
+                  # | p-storage     | task-storage     | Storage       | storage repo  |
+                  # | p-cluster     | task-cluster     | Cluster       | cluster repo  |
 
 
         @regression
```
Comment (gajanan-more) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature:86]: Is the epic number still required?

File: frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: demo-pipeline-to-build-an-image-without-resources
+spec:
+  pipelineRef:
+    name: demo-pipeline-to-build-an-image
+  serviceAccountName: 'default'
+---
+
```
Comment (gajanan-more) [inline frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml:112]: No need to add this empty line. We can remove it

File: frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      resources:
+        inputs:
+          - name: git-repo
+            resource: source-repo
+        outputs:
+          - name: built-image
+            resource: web-image
+
+---
+
```
Comment (gajanan-more) [inline frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml:102]: No need to add this empty line. We can remove it

File: frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  steps:
+    - name: build-an-image
+      image: gcr.io/kaniko-project/executor:v1.3.0
+      command:
+        - /kaniko/executor
+      args:
+        - --dockerfile=$(params.DOCKERFILE)
+        - --destination=$(resources.outputs.built-image.url)
+---
+
```
Comment (gajanan-more) [inline frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml:61]: No need to add this empty line. We can remove it

File: frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+spec:
+  resources:
+    - name: built-image
+      type: image
+      optional: true
+  check:
+    image: alpine
+    script: 'test ! -z $(resources.built-image.url)'
+---
+
```
Comment (gajanan-more) [inline frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml:33]: No need to add this empty line. We can remove it

**Review by gajanan-more — CHANGES_REQUESTED** [review]

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature (hunk 1 above)
Comment (makambalaji) [reply frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature:67]: There is no appropriate test data, so verification is not possible for this particular scenario. so marked them as "@un-verified"

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/pipelines-plugin/integration-tests/features/pipelines/create-from-builder-page.feature:86]: Yes, to show the user still can go with these options but we lack appropriate test data

File: frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml (hunk 3 above)
Comment (sanketpathak) [reply frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml:112]: Updated

Comment (sanketpathak) [reply frontend/packages/pipelines-plugin/integration-tests/testData/pipelines-workspaces/demo-optional-resources.yaml:112]: Updated

**Review by gajanan-more — APPROVED** [review]
/lgtm

**Pull request merged**